rust/catalyst-types/src/id_uri/key_rotation.rs

```diff
@@ -30,6 +30,21 @@ impl KeyRotation {
     pub fn is_default(self) -> bool {
         self == Self::DEFAULT
     }
+
+    /// Get the current rotation from the provided indexed data, if present.
+    pub fn get_key<'a, T>(&self, keys: &'a [T]) -> Option<&'a T> {
+        keys.get(self.0 as usize)
+    }
+
+    /// Get the current rotation from the provided indexed data, if present.
+    pub fn from_keys_decrement<T>(keys: &[T]) -> Self {
```

Comment:
```suggestion
    /// Get the latest rotation value calculated from the provided keys slice (`keys.len() - 1`).
    pub fn latest_rotation<T>(keys: &[T]) -> Self {
```

Reply:
Maybe it's also worth adding a comment about the behaviour with an empty slice.